### ISSUE TITLE
Build 1841: Turtle loses it label, pickaxe, fuel and program

### ISSUE THREAD

**Issue by geoffcorey** [MinimallyCorrect/TickThreading#678]
After 44 blocks of movement the turtle loses it label, pickaxe, fuel and program

while true do
    sleep(0.5)
    turtle.forward()
end

**Comment by LunNova** [comment]
Did a warning about fixing entity list discrepancies pop up in console when this happened? Was it on a chunk boundary when this happened?

**Comment by doctruscott** [comment]
this is the same thing that happens to me, im on 1840 chunks are loaded with chunk loaders and it still loses its items and reverts to a regular turtle and stops moving.

**Comment by geoffcorey** [comment]
What doc said.  Chunks were loaded.

**Comment by JFDreyer** [comment]
Hi there, We are running a server with the FTB ultimate pack 1.1.2.
We have added to the server: Forge Essentials build 272, BechillTPS 0.4.1.1 & now TT #1935.
I run a test server on my machine local that mirrors the above but I added:
Better dungeons 1.4.7 & DrZharks MoCreatures 4.7.0(CustomMobSpawner1.11.2)

Had 3 turtles reset on me today.
I upgraded to TT #1943
I then setup a 4 turtle each to their own chunk to test and then let them mine.
I then came back to find that one of them got reset:
He was traveling east and had reached the border of the chunk assigned.

X: 47 c: 2 (15)
Z: 224 c:14 (0)
Y: 73

Server console showed the following:
2013-06-22 14:37:56 [SEVERE] [TickThreading] Found and fixed 1 discrepancies in tile/entity lists in Overworld/0
tiles - invalid: 1, missing: 0, duplicate: 0, unloaded: 0
entities -  missing: 0, duplicate: 0, unloaded: 0
Took 3ms
2013-06-22 14:49:31 [SEVERE] [TickThreading] Found and fixed 1 discrepancies in tile/entity lists in Overworld/0
tiles - invalid: 1, missing: 0, duplicate: 0, unloaded: 0
entities -  missing: 0, duplicate: 0, unloaded: 0
Took 11ms
2013-06-22 15:02:29 [SEVERE] [TickThreading] Found and fixed 1 discrepancies in tile/entity lists in Overworld/0
tiles - invalid: 1, missing: 0, duplicate: 0, unloaded: 0
entities -  missing: 0, duplicate: 0, unloaded: 0
Took 17ms
2013-06-22 15:16:48 [SEVERE] [TickThreading] Found and fixed 1 discrepancies in tile/entity lists in Overworld/0
tiles - invalid: 1, missing: 0, duplicate: 0, unloaded: 0
entities -  missing: 0, duplicate: 0, unloaded: 0
Took 5ms
2013-06-22 15:35:22 [SEVERE] [TickThreading] Found and fixed 1 discrepancies in tile/entity lists in Overworld/0
tiles - invalid: 1, missing: 0, duplicate: 0, unloaded: 0
entities -  missing: 0, duplicate: 0, unloaded: 0
Took 10ms
2013-06-22 16:03:49 [SEVERE] [TickThreading] Found and fixed 2 discrepancies in tile/entity lists in Overworld/0
tiles - invalid: 1, missing: 0, duplicate: 0, unloaded: 1
entities -  missing: 0, duplicate: 0, unloaded: 0
Took 5ms
2013-06-22 16:48:55 [SEVERE] [TickThreading] Found and fixed 2 discrepancies in tile/entity lists in Overworld/0
tiles - invalid: 1, missing: 0, duplicate: 0, unloaded: 1
entities -  missing: 0, duplicate: 0, unloaded: 0
Took 20ms

Server is still running no crash or chunk unloaded. I was logged into the area the entire time.

I hope this information is helpfull to you.

**Comment by LunNova** [comment]
It looks like the TE discrepancy fixer causes problems if it runs exactly
when a turtle moves. Will make it delay a tick and recheck synchronously.
May need to wait a day for the fix, just arrived on family holiday and they
will probably expect me to have "fun".

**Comment by JFDreyer** [comment]
Cool man, thanks for the quick responce! Also enjoy your holiday!

**Comment by LunNova** [comment]
Can't get this to happen in testing, can you confirm that the fix works?

**Comment by JFDreyer** [comment]
I'll update and repeat the test.

Ok repeated test and didn't loose one turtle, also setup a frame ship to test if the same issues on it have been solved neither of them had and blocks disapear or anything.

From my side this looks fixed in build #1966.

**Comment by ZigTheHedge** [comment]
Build 179, allowWorldUnloading and cleanWorlds both set to false. Issue is happening again. Logs saying:

[SEVERE] [TickThreading] Removed c dan200.turtle.shared.TileEntityTurtle @1838309150 in Digger/Nexus/28 at x, y, z: -79, 57, -71 as it should have been unloaded.

I have 3 turtles digging a single chunk each (chunks loaded). During periodical checks I found one of them has stopped. When I saw her - there was a usual turtle instead of mining one without any programs or inventory in it.
